content/docs/10.configuration-guide/kafka.md

```diff
@@ -179,4 +179,27 @@ For example, for the topic storing executions, you can configure the retention v
       execution:
         properties:
           retention.ms: "86400000"
-```
\ No newline at end of file
+```
+
+## Protecting against too big messages
+
+::alert{type="info"}
+Note: this is an experimental feature.
+::
+
+Kafka topic has a limit of the size of messages it can handle. By default, we set this limit to 10MiB.
+If a message exceeds this limit, it may crash the executor and Kestra will stop.
+
+To prevent that, you can configure a functionality that will automatically store too big messages to the internal storage.
+
+The following configuration will store all messages that exceed 1MiB to the internal storage, and refuse messages that exceed 10MiB by failing the execution, the executor will not crash, and Kestra will keep functioning.
+
+```yaml
+kestra:
+  kafka:
+    message-protection:
+      soft-limit: 1048576
+      hard-limit: 10485760
+```
```

Comment:
```suggestion
To prevent that, you can configure a functionality that will automatically store too big messages to the internal storage.

```yaml
kestra:
  kafka:
    message-protection:
      enabled: true
```

By default, it will store messages that exceed 50% of the topic max message size to the internal storage.

This default is configuration, the following configuration will store all messages that exceed 1MiB to the internal storage, and refuse messages that exceed 10MiB by failing the execution, the executor will not crash, and Kestra will keep functioning.

```yaml
kestra:
  kafka:
    message-protection:
      enabled: true
      soft-limit: 1048576
      hard-limit: 10485760
```
```